src/Providers/ComposerEnvironmentProvider.php

```diff
@@ -8,13 +8,13 @@
 
 class ComposerEnvironmentProvider extends AbstractServiceProvider
 {
-    public function boot(Filesystem $filesystem)
+    public function register()
```

Comment:
Your filesystem is resolved when the app is booted, my implemented makes it resolved only when the ComposerEnvironment is being resolved from IoC.